File: pyg_sellergoods_interface/src/main/java/com/pyg/sellergoods/service/BrandService.java
package com.pyg.sellergoods.service;

import com.pyg.pojo.TbBrand;
import entity.PageResult;
import entity.Result;

import java.util.List;
import java.util.Map;

public interface BrandService {
    /**
     * 查询所有品牌
     * @return
     */
    public List<TbBrand> findAll();
    /**
     * 分页
     */
    public PageResult findPage(int pageNum,int pageSize);
    /**
     * 添加
     */
    public void add(TbBrand tbBrand);
    /**
     * 修改
     */
    public void update(TbBrand tbBrand);
    /**
     * 根据id获取单个品牌信息
     */
    public TbBrand findOne(Long id);
    /**
     * 批量删除
     */
    public void delete(Long[] ids);
    /**
     * 查询分页
     */
    public PageResult findPage(TbBrand tbBrand,int pageNum,int pageSize);

    List<Map> selectOptionList();

}

File: pyg_manage_web/src/main/webapp/js/service/brandService.js
app.service('brandService',function ($http) {
    this.findAll=function () {
        return $http.get('../brand/findAll.do');
    }
    this.findPage=function (page,rows) {
        return $http.get('../brand/findPage.do?page='+page+'&rows='+rows)
    }
    this.findOne=function (id) {
        return $http.get('../brand/findOne.do?id='+id)
    }
    this.dele=function (ids) {
        return $http.get('../brand/delete.do?ids='+ids)
    }
    //保存
    this.add=function (entity) {
        return $http.post('../brand/add.do',entity)
    }
    this.update=function (entity) {
        return $http.post('../brand/update.do',entity)
    }
    this.search=function (page,rows,searchEntity) {
        return $http.post('../brand/search.do?page='+page+'&rows='+rows,searchEntity)
    }
    this.selectOptionList=function () {
        return $http.get("../brand/selectOptionList.do")
    }
});